### PR TITLE
Release 6.0.0

### PR DESCRIPTION
**Pull request by cyattilakiss**
**Description**

This is a major release because we are introducing some backwards compatibility breaking changes in the changes:
#643 [PW 2022] Fix payment-information interceptor to work with other plugins
#656 [PW-2033]: Update Magento2 to latest library version dependency (6.0.0)

Another major release 7.0.0 is planned to release the changes of the improved admin configuration page UI: https://github.com/Adyen/adyen-magento2/tree/feature/admin_ui

Other changes:
#648 [PW-2074] Hide Apple Pay in unsupported browsers 
#653 [PW-2073] Pass the correct store id from order for the modification requests
Fixing Notifications Overview URL
#657 [PW-2150] Fix 3DS2 challenge popup styling 
#660 [PW-2154]: Fixing shopper_locale setup 
#661 Change required magento/module-paypal version to >=100.2.6 
#663 [PW-2005] Fix klarna discount line and logo 
#666 Support combo cards for Mexico & Mexican Pesos
#665 [PW-2173]: Added recurring model to ApplePay BuilderComposite virtual type 
#667 Use quote retrieved by quote id 

**Tested scenarios**

**Fixed issue**:  
#590 savePaymentInformationAndPlaceOrder plugins incorrectly changing return type 
#632 Do not show Apple Pay in unsupported browsers
#494 Wrong store id used for config settings, wrong api url is getting loaded. Related credit memo sync 
#658 Wrong magento/module-paypal require version? 
#664 Payment Methods do not match cart total amount 
#546 Incorrect grand total amount when using REST API

### CODE REVIEW

File: Gateway/Request/CancelDataBuilder.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ public function build(array $buildSubject)
             "reference" => $order->getOrderIncrementId(),
             "originalReference" => $pspReference
         ];
-
+        $request['clientConfig']=[ "storeId" => $payment->getOrder()->getStoreId()];
```
Comment (Aleffio) [inline Gateway/Request/CancelDataBuilder.php]:
```suggestion
        $request['clientConfig'] = ["storeId" => $payment->getOrder()->getStoreId()];
```

File: composer.json [hunk 2]
```diff
@@ -14,10 +14,10 @@
     }
   ],
   "require": {
```
Comment (Aleffio) [inline composer.json]: please update "version" on line 5

File: Gateway/Http/Client/TransactionCapture.php [hunk 3]
```diff
@@ -30,32 +30,19 @@
...
-        \Magento\Framework\Encryption\EncryptorInterface $encryptor,
-        \Adyen\Payment\Helper\Data $adyenHelper,
-        \Adyen\Payment\Logger\AdyenLogger $adyenLogger,
-        \Adyen\Payment\Model\RecurringType $recurringType,
-        array $data = []
+        \Adyen\Payment\Helper\Data $adyenHelper
     ) {
-        $this->_encryptor = $encryptor;
-        $this->_adyenHelper = $adyenHelper;
-        $this->_adyenLogger = $adyenLogger;
```
Comment (Aleffio) [inline Gateway/Http/Client/TransactionCapture.php]: if this is removed, we don't need the one in line 63, either remove both, or readd this one

Comment (cyattilakiss) [reply Gateway/Http/Client/TransactionCapture.php]: Nice catch I wonder why we didn't get any error during the e2e testing.

**Pull request merged**